Refactor grepEntries.py

Nateowami opened the pull request:
Major refactoring in `grepEntries.py`.

Also made `getDependencies.sh` executable.

I intend to add tests in a another PR.

Nateowami committed:
Refactor grepEntries.py

Changes in languagedepot-metadata.sql insignificant and result from
modifications in grepEntries.py because it is a generated file.

Nateowami committed:
Make getDependencies.sh executable

megahirt reviewed (approved):
Looks good.  I will give you some more data to run the code on.

Nateowami committed:
Add basic test with unvalidated data

Test passes, but the values its check against were generated by itself.
Hence, by definition, the test will pass. At this point it is necessary
to check manually that the values it generated are correct.

Nateowami committed:
grep even if not all directories exist in project

Nateowami commented:
I mistakenly thought this had been merged, so I pushed two more commits. If you like I can try to move them to another branch. I'm still verifying the test results of the first commit (all looks good so far), but the second one (ca303ad) is in error.

Nateowami committed:
Properly convert None to NULL for Postgres

Nateowami committed:
Fix non-determinism in xmlEntries.py

Nateowami committed:
Specify sort by id when exporting as CSV

Nateowami commented:
The last three commits are all small bug fixes. Now when I run the script on a dataset of lots of projects, I consistently get the same output. Formerly the output was nondeterministic.
